ambry-frontend/src/main/java/com.github.ambry.frontend/AmbryBlobStorageServiceFactory.java

```diff
@@ -57,15 +60,19 @@ public AmbryBlobStorageServiceFactory(VerifiableProperties verifiableProperties,
     if (verifiableProperties == null || clusterMap == null || responseHandler == null || router == null) {
       throw new IllegalArgumentException("Null arguments were provided during instantiation!");
     } else {
+      MetricRegistry metricRegistry = clusterMap.getMetricRegistry();
       frontendConfig = new FrontendConfig(verifiableProperties);
-      frontendMetrics = new FrontendMetrics(clusterMap.getMetricRegistry());
+      frontendMetrics = new FrontendMetrics(metricRegistry);
       this.clusterMap = clusterMap;
       this.responseHandler = responseHandler;
       this.router = router;
+      AclServiceFactory aclServiceFactory =
+          Utils.getObj(frontendConfig.frontendAclServiceFactory, verifiableProperties, metricRegistry);
+      AclService aclService = aclServiceFactory.getAclService();
       idConverterFactory =
-          Utils.getObj(frontendConfig.frontendIdConverterFactory, verifiableProperties, clusterMap.getMetricRegistry());
+          Utils.getObj(frontendConfig.frontendIdConverterFactory, verifiableProperties, metricRegistry);
       securityServiceFactory = Utils.getObj(frontendConfig.frontendSecurityServiceFactory, verifiableProperties,
-          clusterMap.getMetricRegistry());
+          clusterMap.getMetricRegistry(), aclService);
```

Comment:
I agree that would be better, since the AclService implementation is usually coupled with a specific SecurityService implementation as well.